### ISSUE TITLE
Critical pods should not be limited to kube-system namespace

### ISSUE THREAD

**Issue by bsalamat** [kubernetes/kubernetes#60596]
When we used annotations to mark critical pods, we considered pods with such an annotation critical only when they were created in "kube-system" namespace. Going to the new era of using priority to indicate critical pods, we would like to remove the restriction of "kube-system" namespace and consider pods with critical priority as critical no matter what namespace they are in.

ref/ https://github.com/kubernetes/kubernetes/pull/60519#discussion_r171132505

/kind bug
/sig scheduling
/assign @ravisantoshgudimetla 
cc/ @liggitt

**Comment by ravisantoshgudimetla** [comment]
While I am fine with the approach, it seems the only way to ensure that those critical pods are not coming from some random namespace is whitelisting them using priority admission controller or using quota. Atleast previously we have one more level of security that critical pod has to be created in kube-system namespace.

**Comment by liggitt** [comment]
> previously we have one more level of security that critical pod has to be created in kube-system namespace

The requirement to limit the critical pod annotation meaning to the kube-system namespace was a workaround for the lack of access control and quota around the annotation. We actively work to prevent special handling of the kube-system namespace wherever possible.

**Comment by liggitt** [comment]
> the only way to ensure that those critical pods are not coming from some random namespace is whitelisting them using priority admission controller or using quota

Yes, which is why those aspects are essential to a functional priority system that works for a variety of cluster configurations

**Comment by derekwaynecarr** [comment]
Agree with Ravi and Jordan on what is needed for final solution.

**Comment by bsalamat** [comment]
There is no doubt that we need to have quota for priority. It is needed not only for critical priorities, but also for any other high priority classes that users may define. That effort is going forward and we will have it soon.

**Comment by k82cn** [comment]
@ravisantoshgudimetla , I think we also need to update Priority admission controller for that.

```
root@wgl-ubuntu:/home/k82cn/go_ws/src/k8s.io/kubernetes# kc create -f pod.yaml
Error from server (Forbidden): error when creating "pod.yaml": pods "rss-site" is forbidden: pods with system-cluster-critical priorityClass is not permitted in default namespace

root@wgl-ubuntu:/home/k82cn/go_ws/src/k8s.io/kubernetes# kc version
Client Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0-alpha.1.439+535426bf736a31-dirty", GitCommit:"535426bf736a31edeb76cf64f3b6d90e6996e49a", GitTreeState:"dirty", BuildDate:"2018-05-01T07:14:55Z", GoVersion:"go1.10.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.0-alpha.0.1200+323e1375b3b15a", GitCommit:"323e1375b3b15a4144d46d41155ca402a2fcef63", GitTreeState:"clean", BuildDate:"2018-09-27T06:52:20Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}

root@wgl-ubuntu:/home/k82cn/go_ws/src/k8s.io/kubernetes# cat pod.yaml
apiVersion: v1
kind: Pod
metadata:
  name: rss-site
  labels:
    app: web
spec:
  priorityClassName: system-cluster-critical
  containers:
    - name: front-end
      image: nginx
      ports:
        - containerPort: 80
```

**Comment by liggitt** [comment]
see later PR re-restricting this until the quota controls graduate to at least beta state: https://github.com/kubernetes/kubernetes/pull/65593

**Comment by k82cn** [comment]
Got that, thanks :) That makes sense to me.

**Comment by llarsson** [comment]
If this is closed and there even was a [merged pull request](https://github.com/kubernetes/kubernetes/pull/60653) for this type of functionality, what is the current status? And is [the documentation](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/) in sync with current status?

**Comment by nevalla** [comment]
It seems that `kube-system` namespace is still [hard coded in Priority plugin](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/priority/admission.go#L143-L155), so I guess the current status is that system critical pods are still limited to `kube-system` namespace.

**Comment by tlvenn** [comment]
@bsalamat can we reopen this issue please because as pointed out by @nevalla, the issue still exists. Thanks.

**Comment by zimmertr** [comment]
Waiting for this to be available myself. No reason my Node Exporter dataset has to run in the `kube-system` namespace instead of my Grafana namespace.

**Comment by vllry** [comment]
Confirmed code + behavior.
/open

**Comment by nesl247** [comment]
> If this is closed and there even was a [merged pull request](https://github.com/kubernetes/kubernetes/pull/60653) for this type of functionality, what is the current status? And is [the documentation](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/) in sync with current status?

I too am confused based on that documentation. Can we get an update here with an answer and fix the documentation if it's wrong?

**Comment by liggitt** [comment]
This is resolved in 1.17

See https://github.com/kubernetes/kubernetes/pull/76310
